src/deadline_test_fixtures/deadline/worker.py

```diff
@@ -619,6 +629,41 @@ def ami_id(self) -> str:
 
         return self._ami_id
 
+    def _wait_until_userdata_finishes(self) -> tuple[bool, str]:
+        result: CommandResult | None = None
+        success: bool = False
+        LOG.info("Waiting for userdata to finish")
+
+        def get_userdata_result() -> bool:
+            nonlocal result
+            nonlocal success
+            result = self.send_command(self.userdata_success_script())
+
+            if self.USERDATA_SUCCESS_STRING in str(result):
+                success = True
+                return True
+
+            if self.USERDATA_FAILURE_STRING in str(result):
+                success = False
+                return True
+
+            return False
+
+        # Raises TimeoutError if the userdata status cannot be fetched in
+        # the given timeframe.
+        wait_for(
+            description="getting the result of userdata",
+            predicate=get_userdata_result,
+            interval_s=5,
+            max_retries=60,
+        )
+
+        LOG.info(
+            "Userdata finished %s.",
+            "successfully" if success else "unsuccessfully",
+        )
```

Comment:
nit: it seems like we could use `USERDATA_SUCCESS_STRING` / `USERDATA_FAILURE_STRING` here too